util/linuxfw/iptables.go

```diff
@@ -823,3 +825,479 @@ type markTarget struct {
 	Mode markMode
 	Mark uint32
 }
+
+type iptablesInterface interface {
+	// Adding this interface for testing purposes so we can mock out
+	// the iptables library, in reality this is a wrapper to *iptables.IPTables.
+	Insert(table, chain string, pos int, args ...string) error
+	Append(table, chain string, args ...string) error
+	Exists(table, chain string, args ...string) (bool, error)
+	Delete(table, chain string, args ...string) error
+	ClearChain(table, chain string) error
+	NewChain(table, chain string) error
+	DeleteChain(table, chain string) error
+}
+
+type iptableRunner struct {
```

Comment:
s/iptableRunner/iptablesRunner

Reply:
Sorry are you suggesting we should go with iptablesRunner?

Reply:
Updated